src/main/java/com/wavefront/opentracing/WavefrontTracer.java

```diff
@@ -266,18 +267,20 @@ void reportWavefrontGeneratedData(WavefrontSpan span) {
       put(COMPONENT_TAG_KEY, span.getComponentTagValue());
     }};
 
-    boolean customTagMatch = false;
     // avoid iterating through the span tags if user has not instantiated any red-metric custom tags
-    if (redMetricsCustomTagKeys.size() > 0) {
+    if (redMetricsCustomTagKeys.size() > 1) {
       Map<String, Collection<String>> spanTags = span.getTagsAsMap();
       for (String customTagKey : redMetricsCustomTagKeys) {
         if (spanTags.containsKey(customTagKey)) {
-          customTagMatch = true;
           // Assuming at least one value exists ...
           pointTags.put(customTagKey, spanTags.get(customTagKey).iterator().next());
         }
       }
     }
+    else {
+      // span.kind tag will be promoted by default
+      pointTags.put(SPAN_KIND.getKey(), NULL_TAG_VAL);
```

Comment:
the span still might have `span.kind` tag in this else clause

Reply:
same above